### PR TITLE
Project lib integrations - VScode

### PR DESCRIPTION
**Pull request by ehysi**
- Add support for project libs
- Add rename/move commands to plugin content

### CODE REVIEW

**Review by StanislasBertrand — APPROVED** [review]
Works well and code LGTM ! Maybe @apichery will want to take a look as well

**Comment by apichery** [comment]
Where do the icon comes from? Do we have rights to redistribute it?

**Comment by ehysi** [comment]
> Where does the icon comes from? Do we have rights to redistribute it?

It comes from https://www.svgrepo.com/svg/229807/library-book . The license is CC0 which is free for commerce and redistribution. Basically a public domain license.

File: src/FSManager.ts [hunk 1]
```diff
@@ -30,10 +30,10 @@ export class FileDetails {
         return new FileDetails(fileName, content, directory);
     }
 
-    static fromLibrary(pluginId: string, filePath: string, content?: string): FileDetails {
+    static fromLibrary(projectKey: string, filePath: string, content?: string): FileDetails {
         const libraryPath = filePath.substr(0, filePath.lastIndexOf('/'));
         const fileName = filePath.substr(filePath.lastIndexOf('/') + 1);
-        let directory = "LIBRARIES/" + pluginId;
+        let directory = "LIBRARIES/" + projectKey;
```
Comment (apichery) [inline src/FSManager.ts:36]: `LIBRARIES`=>`lib`

**Review by apichery — CHANGES_REQUESTED** [review]

**Pull request merged**